src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +462,127 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
         return this;
     }
 
+    /**
+     * Use {@link #withFIPSEnabled(Path)} with default value of {@code target/dependency}
+     */
+    public RealJenkinsRule withFIPSEnabled() {
+        return withFIPSEnabled(Path.of("target", "dependency"));
+    }
+
+    /**
+     * Enable FIPS mode assuming all FIPS BouncyCastle jars (bc-fips.jar, bctls-fips.jar and bcpkix-fips.jar) are in the provided directory.
+     * This will set some System properties such :
+     * <ul>
+     *   <li>-Xbootclasspath with FIPS BouncyCastle jars</li>
+     *   <li>java.security.properties to a new temp file configuring the security provider to FIPS BouncyCastle</li>
+     *   <li>-Dsecurity.overridePropertiesFile=true</li>
+     *   <li>-Djavax.net.ssl.trustStoreType=PKCS12</li>
+     *   <li>-Djenkins.security.FIPS140.COMPLIANCE=true</li>
+     *   <li>-Dcom.redhat.fips=false</li>
+     * </ul>
+     * @param fipsLibrariesPath path to directory with FIPS BouncyCastle jars
+     */
+    public RealJenkinsRule withFIPSEnabled(@NonNull Path fipsLibrariesPath) {
+        this.fipsLibrariesPath = Objects.requireNonNull(fipsLibrariesPath);
+        // check expected BouncyCastle fips jars are here
+        if  (!Files.exists(this.fipsLibrariesPath) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bc-fips.jar")) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bctls-fips.jar")) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bcpkix-fips.jar"))) {
+            throw fipsIllegalSetupException(this.fipsLibrariesPath);
+        }
+
+        withBootClasspath(fipsLibrariesPath.resolve("bc-fips.jar").toFile(),
+                            fipsLibrariesPath.resolve("bctls-fips.jar").toFile(),
+                            fipsLibrariesPath.resolve("bcpkix-fips.jar").toFile());
+        try {
+            // please note == is not a typo, but it makes our file completely override the jvm security file
+            javaOptions(    "-Djava.security.properties==" + writeFIPSJavaSecurityFile().toUri(),
```

Comment:
you have some funky indentation here!  (the following lines may also need adjusting if spotless complains
```suggestion
            javaOptions("-Djava.security.properties==" + writeFIPSJavaSecurityFile().toUri(),
```

Reply:
fixed.
but frankly would be better to have spotless activated (I will propose it once this one is merged)
we have plenty more other interesting things to do in our life than discussing about spaces :)